### PR TITLE
Shopping Cart

### PR DESCRIPTION
**Pull request by meta-akanksha-khurana**

### CODE REVIEW

File: Fundamental/Assignment1/src/Shopping.java [hunk 1]
```diff
@@ -0,0 +1,152 @@
+/**
+ * 
+ *      Assignment 1
```
Comment (meta-dharmveer-kirodiwal) [inline Fundamental/Assignment1/src/Shopping.java:3]: Commenting should be more specific and clear

File: Fundamental/Assignment1/src/Shopping.java [hunk 2]
```diff
@@ -0,0 +1,152 @@
...
+	
+	Shopping()
+	{
+		add(new Item("Egg",20.00,30));
+		add(new Item("Bread",30.00,10));
+		add(new Item("Butter",40.00,20));	
+	}
+	
+	//Function to add item in the store
+	void add(Item i)
```
Comment (meta-dharmveer-kirodiwal) [inline Fundamental/Assignment1/src/Shopping.java:30]: Parameter name should be understandable. like "item"

File: Fundamental/Assignment1/src/Shopping.java [hunk 3]
```diff
@@ -0,0 +1,152 @@
...
+			{
+				total+=m.getValue()*items.get(m.getKey()).price;
+			}
+			System.out.println("Total Bill:"+total);
+		}
+	}
+	
+	
+	public static void main(String[] args) {
+		Shopping.Cart c=new Shopping().new Cart();
```
Comment (meta-dharmveer-kirodiwal) [inline Fundamental/Assignment1/src/Shopping.java:103]: Object and variable name should be clear.

File: Fundamental/Assignment1/src/Shopping.java [hunk 4]
```diff
@@ -0,0 +1,152 @@
...
+			display();
+			System.out.println("Enter the id of item:");
+			int id=sc.nextInt();
+			System.out.println("Enter quantity of item:");
+			int quantity=sc.nextInt();
+			purchaseItem.put(id,quantity);
+			items.replace(id,new Item(items.get(id).Name,items.get(id).price,items.get(id).stock-quantity ));
+		}
+		
+		//Function to delete item from the cart
```
Comment (meta-dharmveer-kirodiwal) [inline Fundamental/Assignment1/src/Shopping.java:62]: Use proper commenting to describe method functionality,parameter and exceptions.

File: Fundamental/Assignment1/src/Shopping.java [hunk 5]
```diff
@@ -0,0 +1,152 @@
...
+	
+	//Function to display items in store
+	void display()
+	{
+		System.out.println("ID"+" "+"Name"+" "+"Quantity");
+		for(Map.Entry<Integer, Item> m:items.entrySet())
+			System.out.println(m.getKey()+" "+m.getValue().Name+" "+m.getValue().price+" "+m.getValue().stock);
+	}
+
+	class Cart
```
Comment (meta-dharmveer-kirodiwal) [inline Fundamental/Assignment1/src/Shopping.java:44]: Try to make classes separate if those are not more private and dependent.

File: Fundamental/Assignment1/src/Shopping.java [hunk 6]
```diff
@@ -0,0 +1,152 @@
+/**
+ * 
+ *      Assignment 1
+ */
+import java.util.*;
+class Item{
+	String Name;
```
Comment (meta-dharmveer-kirodiwal) [inline Fundamental/Assignment1/src/Shopping.java:7]: Use proper naming convention to define variables.

File: Fundamental/Assignment1/src/Shopping.java (hunk 1 above)
Comment (meta-akanksha-khurana) [reply Fundamental/Assignment1/src/Shopping.java:3]: I will do this in future assignments and will follow the coding conventions which are given in the pdf.

File: Fundamental/Assignment1/src/Shopping.java (hunk 2 above)
Comment (meta-akanksha-khurana) [reply Fundamental/Assignment1/src/Shopping.java:30]: will keep in mind in future assignments

File: Fundamental/Assignment1/src/Shopping.java (hunk 3 above)
Comment (meta-akanksha-khurana) [reply Fundamental/Assignment1/src/Shopping.java:103]: will  follow it in future assignments

File: Fundamental/Assignment1/src/Shopping.java (hunk 4 above)
Comment (meta-akanksha-khurana) [reply Fundamental/Assignment1/src/Shopping.java:62]: will follow the convention given in the guidelines.

File: Fundamental/Assignment1/src/Shopping.java (hunk 5 above)
Comment (meta-akanksha-khurana) [reply Fundamental/Assignment1/src/Shopping.java:44]: okay..I will keep this in mind

File: Fundamental/Assignment1/src/Shopping.java (hunk 6 above)
Comment (meta-akanksha-khurana) [reply Fundamental/Assignment1/src/Shopping.java:7]: I will keep in mind in future assignments

**Review by meta-dharmveer-kirodiwal — APPROVED** [review]

**Pull request merged**